sandy/fy.py

```diff
@@ -445,24 +438,21 @@ def gls_update(self, zam, e, decay_data, xp, Vy=None, kind='cumulative',
 
         """
         data = self.data
-        S = _gls_setup(decay_data, kind, index_columns)
         # Filter FY data:
         fy_data = self.filter_by('ZAM', zam)\
                       .filter_by("E", e).data\
                       .set_index('ZAP')[['MT', 'FY', 'DFY']]
         # Divide the data:
         fy_type = fy_data.groupby('MT')
         IFY, CFY = [fy_type.get_group(x)['FY'] for x in fy_type.groups]
-        DIFY, DCFY = [fy_type.get_group(x)['DFY'] for x in fy_type.groups]
+        mask = (data.ZAM == zam) & (data.MT == 454) & (data.E == e)
+        DIFY = data.loc[mask, 'DFY']
```

Comment:
here you are filtering again...but everything was already filtered in `fy_data`.

Couldn't you do `DIFY = fy_data.query('MT==459).DFY` ?